### ISSUE TITLE
Automatically strip/ignore domain in login form

### ISSUE THREAD

**Issue by jpawlowski** [opnsense/core#3749]
I understand it is essentially needed that a username may not contain any characters that can't be used in a Unix login name. This would include the @ sign, being part of an email address.

When using (Azure) Active Directory authentication, the parameter `User naming attribute` cannot be set to `userPrincipalName` (UPN) but would need to stay with `sAMAccountName`.

However, living in the Azure AD Hybrid and Office 365 world, end-users have been told to use their e-mail address to login (which in turn should always match the `userPrincipalName` for that reason, to ensure identity related featured like [AIP](https://docs.microsoft.com/azure/information-protection/) will work properly).
Also, `sAMAccountName` will likely be the same as the e-mail prefix today with only old accounts having a last century so called "pre-Windows 2000 login" in a cryptic naming convention a company had defined in the world before e-mail.

**TL;DR**
Long story short: End-users who would like to login to OPNsense to consume some self-service applications (e.g. download OpenVPN configuration - see #3748) should be able to use their Active Directory UPN (e-mail address) to login.
The idea is to let the user enter any e-mail address but simply strip the `@domain.tld` part before further processing. This is not a big change but make end-user communication much easier as you don't need to train users when to use their full e-mail address and when to leave out the domain part.

Surely this would not allow any multi-tenant setups by handling domains differently. However, OPNsense itself is not fully multi-tenant capable anyway, due to mutually exclusive Unix user names.

**Comment by AdSchellevis** [comment]
This issue has been automatically timed-out (after 180 days of inactivity).

For more information about the policies for this repository,
please read https://github.com/opnsense/core/blob/master/CONTRIBUTING.md for further details.

If someone wants to step up and work on this issue,
just let us know, so we can reopen the issue and assign an owner to it.
